tests/helpers/reporter/reporter.go

```diff
@@ -0,0 +1,211 @@
+
+	//Write JSON for code Coverage
+	codeCoverageRegex := metric.CodeRegex{
+		LogFile: LogPath + "test_summary.log",
+		//	JsonFile:
+		LineCoverageField:     "Lines Coverage: ",
+		FunctionCoverageField: "Functions Coverage: ",
+	}
+
+	codeCoverageResult := metric.Code{
+		Name: "code_coverage",
+		Type: "Code coverage",
+	}
+
+	codeCoverage :=
+		metric.GetCodeCoverage(codeCoverageRegex, codeCoverageResult)
+	codeCoverage.WriteJSONCodeCoverage()
+}
+
+func writeReturnCode() {
+
+	//Return codes are stored in the test_summary.log
+
+	var LogFile = LogPath + "test_summary.log"
+	const (
+		CppcheckRE         = "cppcheck return code: "
+		CppcheckName       = "ccpcheck"
+		AutogenRE          = "autogen return code: "
+		AutogenName        = "autogen"
+		MakeRE             = "make return code: "
+		MakeName           = "make"
+		MakeInstallRE      = "make install return code: "
+		MakeInstallName    = "make_install"
+		ProxyTestName      = "check_proxy"
+		ProxyTestRE        = "check-proxy return code: "
+		UnitTestName       = "unit_test"
+		UnitTestRE         = "unit tests return code: "
+		FunctionalTestRE   = "functional tests return code: "
+		FunctionalTestName = "functional_test"
+		ValgrindName       = "valgrind"
+		ValgrindRE         = "valgrind tests return code: "
+		CoverageName       = "coverage"
+		CoverageRE         = "coverage return code: "
+		DockerTestName     = "docker_test"
+		DockerTestRE       = "docker tests return code: "
+	)
+
+	//Write ccpcheck JSON file
+	cppcheck :=
+		metric.GetJSONReturnCode(LogFile, CppcheckName, CppcheckRE)
+	cppcheck.WriteJSONReturnCode()
+
+	//Write autogen JSON file
+	autogen := metric.GetJSONReturnCode(LogFile, AutogenName, AutogenRE)
+	autogen.WriteJSONReturnCode()
+
+	//Write make JSON file
+	make := metric.GetJSONReturnCode(LogFile, MakeName, MakeRE)
+	make.WriteJSONReturnCode()
+
+	//Write make install JSON file
+	makeInstall :=
+		metric.GetJSONReturnCode(
+			LogFile, MakeInstallName, MakeInstallRE)
+	makeInstall.WriteJSONReturnCode()
+
+	//Write Proxy Test JSON file
+	proxyTest :=
+		metric.GetJSONReturnCode(LogFile, ProxyTestName, ProxyTestRE)
+	proxyTest.WriteJSONReturnCode()
+
+	//Write Unit Test Name JSON file
+	unitTest :=
+		metric.GetJSONReturnCode(LogFile, UnitTestName, UnitTestRE)
+	unitTest.WriteJSONReturnCode()
+
+	//Write Functional Test JSON file
+	functionalTest :=
+		metric.GetJSONReturnCode(
+			LogFile, FunctionalTestName, FunctionalTestRE)
+	functionalTest.WriteJSONReturnCode()
+
+	//Write Valgrind JSON file
+	valgrind :=
+		metric.GetJSONReturnCode(LogFile, ValgrindName, ValgrindRE)
+	valgrind.WriteJSONReturnCode()
+
+	//Write Code Coverage JSON file
+	coverage :=
+		metric.GetJSONReturnCode(LogFile, CoverageName, CoverageRE)
+	coverage.WriteJSONReturnCode()
+
+	//Write Docker JSON file
+	docker :=
+		metric.GetJSONReturnCode(LogFile, DockerTestName, DockerTestRE)
+	docker.WriteJSONReturnCode()
+
+}
+
+func main() {
+
+	LogPath = "../test_logs/"
```

Comment:
Yes, you most definitely want an option so this program can be driven by an external script generating the logs.